orc8r/tools/helm/package.sh

```diff
@@ -47,78 +65,143 @@ if [ "$DEPLOYMENT_TYPE" != "$FWA" ] && [ "$DEPLOYMENT_TYPE" != "$FFWA" ] && [ "$
   ['$FWA', '$FFWA', '$ALL']"
 fi
 
-if [[ -z $GITHUB_REPO ]]; then
-  exitmsg "Environment variable GITHUB_REPO must be set"
-fi
-
-if [[ -z $GITHUB_REPO_URL ]]; then
-  exitmsg "Environment variable GITHUB_REPO_URL must be set"
-fi
-
-if [[ -z $GITHUB_USERNAME ]]; then
-  exitmsg "Environment variable GITHUB_USERNAME must be set"
-fi
-
-if [[ -z $GITHUB_ACCESS_TOKEN ]]; then
-  exitmsg "Environment variable GITHUB_ACCESS_TOKEN must be set"
-fi
-
-if [[ -z $MAGMA_ROOT ]]; then
-  exitmsg "Environment variable MAGMA_ROOT must be set"
-fi
+# Check for artifactory URL presence
+if [[ -z $HELM_CHART_MUSEUM_URL ]]; then
 
-# Set up repo for charts
-mkdir -p ~/magma-charts && cd ~/magma-charts
-git init
+  if [[ -z $GITHUB_REPO ]]; then
+    exitmsg "Environment variable GITHUB_REPO must be set"
+  fi
 
-# Begin packaging necessary helm charts
-helm dependency update $MAGMA_ROOT/orc8r/cloud/helm/orc8r/
-helm package $MAGMA_ROOT/orc8r/cloud/helm/orc8r/ && helm repo index .
+  if [[ -z $GITHUB_REPO_URL ]]; then
+    exitmsg "Environment variable GITHUB_REPO_URL must be set"
+  fi
 
-if [ "$DEPLOYMENT_TYPE" == "$FWA" ]; then
-  helm dependency update $MAGMA_ROOT/lte/cloud/helm/lte-orc8r/
-  helm package $MAGMA_ROOT/lte/cloud/helm/lte-orc8r/ && helm repo index .
-fi
+  if [[ -z $GITHUB_USERNAME ]]; then
+    exitmsg "Environment variable GITHUB_USERNAME must be set"
+  fi
 
-if [ "$DEPLOYMENT_TYPE" == "$FFWA" ]; then
-  helm dependency update $MAGMA_ROOT/lte/cloud/helm/lte-orc8r/
-  helm package $MAGMA_ROOT/lte/cloud/helm/lte-orc8r/ && helm repo index .
+  if [[ -z $GITHUB_ACCESS_TOKEN ]]; then
+    exitmsg "Environment variable GITHUB_ACCESS_TOKEN must be set"
+  fi
 
-  helm dependency update $MAGMA_ROOT/feg/cloud/helm/feg-orc8r/
-  helm package $MAGMA_ROOT/feg/cloud/helm/feg-orc8r/ && helm repo index .
+  if [[ -z $MAGMA_ROOT ]]; then
+    exitmsg "Environment variable MAGMA_ROOT must be set"
+  fi
+
+  # Set up repo for charts
```

Comment:
Seems like this script is now packaging *twice* and unconditionally uploading to *both* GitHub and an artifactory. Correct me if I'm missing something

Reply:
On line 69, I am checking for the presence of HELM_CHART_MUSEUM_URL.
If absent, the script will upload to private github repo otherwise it will upload to the defined Artifactory repo.
Thanks for reviewing it in such depth! :)